extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -72,16 +78,35 @@ class AmpYoutube extends AMP.BaseElement {
     iframe.width = this.width_;
     iframe.height = this.height_;
     this.element.appendChild(iframe);
-    /** @private {?Element} */
+
+    /** @private {!Element} */
     this.iframe_ = iframe;
 
-    // TODO(mkhatib, #2050): Use PlayerReady message for layout promise.
-    return loadPromise(iframe);
+    /** @private @const {!Promise} */
+    this.playerReadyPromise_ = this.getPlayerReadyPromise_();
+
+    this.getWin().addEventListener(
```

Comment:
I'm a bit confused on this part. Should "message" listener be set on the current window `getWin()` or on `iframe.contentWindow`? Could you please clarify?